parlai/core/torch_generator_agent.py

```diff
@@ -1052,6 +1094,14 @@ def set_context(self: TSType, context: torch.LongTensor) -> TSType:
         self.context = context.tolist()
         return self
 
+    def set_blacklist(self: TSType, blacklist: List[List[int]]) -> TSType:
+        self.blacklist: Dict[int, List[List[int]]] = {}
+        for ngram in blacklist:
```

Comment:
poss. dumb q... why do we do this `set_blacklist` every time the tree search factory is created? don't we only need to create this dictionary once?

Reply:
hmm, ya you're right. i figured it was a more convenient form to keep it in the agent, in case you wanted to dynamically update it. I don't think the O(n) conversion is that bad tho.

Reply:
i see .. still a bit weird. could create a blacklist object to allow you to dynamically update

Reply:
i agree that we'd want some way to dynamically update the blacklist, though perhaps Emily is right that we'd want to just pass along a shared object rather than rebuild. given that `n` is not gonna be too large here it's prob fine either way